docs/skill-guide.md

```diff
@@ -0,0 +1,230 @@
+The <a href="../scaffolding/">scaffolding tool</a> allows you create the folder structure required for a skill.
+
+!!!	Note
+	Before developing your first skill, please read the <a href="../skill/">skill guide</a>.
+
+## Step 1: Setup
+
+Ensure, you have followed the <a href="../quickstart/">preliminaries</a>. We will first create an agent and add a scaffold skill, which we call `my_search`:
+
+``` bash
+aea create my_agent && cd my_agent
+aea scaffold skill my_search
+```
+
+## Step 2: Develop a Behaviour
+
+A `Behaviour` class contains the business logic specific to initial actions initiated by the agent rather than reactions to other events.
+
+In this example, we implement a simple search behaviour. Each time, `act()` gets called, we will send a search request to the OEF.
+
+``` python
+from aea.protocols.oef.message import OEFMessage
+from aea.protocols.oef.models import Query, Constraint, ConstraintType
+from aea.protocols.oef.serialization import DEFAULT_OEF, OEFSerializer
+from aea.skills.base import Behaviour
+
+
+class MySearchBehaviour(Behaviour):
+    """This class provides a simple search behaviour."""
+
+    def __init__(self, **kwargs):
+        """Initialize the search behaviour."""
+        super().__init__(**kwargs)
+        self.sent_search_count = 0
+
+    def setup(self) -> None:
+        """
+        Implement the setup.
+
+        :return: None
+        """
+        pass
+        
+
+    def act(self) -> None:
+        """
+        Implement the act.
+
+        :return: None
+        """
+        self.sent_search_count += 1
+        search_constraints = [Constraint("country",
+        	                  ConstraintType("==", "UK"))]
+        search_query_w_empty_model = Query(search_constraints, model=None)
+        search_request = OEFMessage(oef_type=OEFMessage.Type.SEARCH_SERVICES,
+                                    id=self.sent_search_count,
+                                    query=search_query_w_empty_model)
+        self.context.outbox.put_message(to=DEFAULT_OEF,
+                                        sender=self.context.agent_address,
+                                        protocol_id=OEFMessage.protocol_id,
+                                        message=OEFSerializer().encode(search_request))
+
+    def teardown(self) -> None:
+        """
+        Implement the task teardown.
+
+        :return: None
+        """
+        pass
+```
+
+Searches are proactive and as such well placed in a `Behaviour`.
+
+
+## Step 3: Develop a Handler
+
+So far, we have tasked the agent with sending search requests to the OEF. However, we have no way of handling the responses sent to the agent by the OEF at the moment. The agent would simply respond to the OEF via the default `error` skill which sends all unrecognized envelopes back to the sender.
+
+Let us now implement a handler to deal with the incoming search responses.
+
```

Comment:
the same issue of above - remind that the code should be located in `<agent_name>/skills/my_search/handlers.py`